_source/_posts/2022-08-15-spring-boot-crud-update.md

```diff
@@ -0,0 +1,1069 @@
+
+`src/Api.js`
+
+```js
+import axios from 'axios'
+
+const instance = axios.create({
+    baseURL: process.env.VUE_APP_SERVER_URI,
+    timeout: 1000
+});
+
+const createApi = (auth) => {
+
+    instance.interceptors.request.use(async function (config) {
+        let accessToken = await auth.getAccessToken()
+        config.headers = {
+            Authorization: `Bearer ${accessToken}`
+        }
+        return config;
+    }, function (error) {
+        return Promise.reject(error);
+    });
+
+    return {
+
+        // (C)reate
+        createNew(text, completed) {
+            return instance.post('/todos', {title: text, completed: completed})
+        },
+
+        // (R)ead
+        getAll() {
+            return instance.get( '/todos', {
+                transformResponse: [function (data) {
+                    return data ? JSON.parse(data)._embedded.todos : data;
+                }]
+            })
+        },
+
+        // (U)pdate
+        updateForId(id, text, completed) {
+            return instance.put('todos/' + id, {title: text, completed: completed})
+        },
+
+        // (D)elete
+        removeForId(id) {
+            return instance.delete('todos/' + id)
+        }
+    }
+}
+
+export default createApi
+```
+
+All of the requests to the server go through this module. Notice that the server URI is hard coded at the top as `SERVER_URL`. In production this would be moved to a config file or an environment variable. Also take a look at how the access token is retrieved from the global `auth` object and injected into every request.
+
+Create the router file.
+
+`src/router/index.js`
+
+```js
+import { createRouter, createWebHistory } from 'vue-router'
+import { navigationGuard } from '@okta/okta-vue'
+import Todos from "@/components/Todos";
+import Home from "@/components/Home";
+import { LoginCallback } from '@okta/okta-vue'
+
+const routes = [
+    {
+        path: '/',
+        component: Home
+    },
+    {
+        path: '/todos',
+        component: Todos,
+        meta: {
+            requiresAuth: true
+        }
+    },
+    { path: '/login/callback', component: LoginCallback },
+]
+
+const router = createRouter({
+    history: createWebHistory(process.env.BASE_URL),
+    routes,
+})
+
+router.beforeEach(navigationGuard)
+
+export default router
+```
+
+The router has three paths. The home path and the todos path are straightforward. The last path, `/login/callback`, is provided by the Okta module and it is what handles the login redirect from the Okta servers after authentication.
+
+Create the `Home` component.
+
+`src/components/Home.vue`
+
+```vue
+<template>
+  <div class="column justify-center items-center" id="row-container">
+    <q-card class="my-card">
+      <q-card-section style="text-align: center">
+        <div v-if='authState && authState.isAuthenticated' >
+          <h6 v-if="claims && claims.email">You are logged in as {{ '{{' }}claims.email}}</h6>
```

Comment:
@mraible This line, for example

Reply:
You should be able to wrap the whole code block with `{% raw %}` and `{% endraw %}`.